src/main/java/com/amam/wizardschool/controller/StudentController.java

```diff
@@ -18,44 +20,67 @@ public StudentController(StudentService studentService) {
     }
 
     @GetMapping("/all")
+    @Operation(summary = "Get all students")
     public Collection<Student> getStudents() {
         return studentService.getStudents();
     }
 
     @GetMapping("{id}")
+    @Operation(summary = "Get student by ID")
     public ResponseEntity<Student> getStudent(@PathVariable("id") Long id) {
         return studentService.findStudent(id).isPresent() ?
                 ResponseEntity.ok(studentService.findStudent(id).get()) :
                 ResponseEntity.badRequest().build();
 
     }
 
-    @GetMapping
-    public ResponseEntity<Collection<Student>> getStudentsByAge(@RequestParam("age") int age) {
+    @GetMapping("/age/{age}")
+    @Operation(summary = "Get students by Age")
+    public ResponseEntity<Collection<Student>> getStudentsByAge(@PathVariable int age) {
         return ResponseEntity.ok(studentService.getStudentsByAge(age));
     }
 
+    @GetMapping("/age")
+    @Operation(summary = "Get students by Age between two values")
+    public ResponseEntity<Collection<Student>> getStudentsByAgeBetween(@RequestParam(required = false, defaultValue = "0") int min,
+                                                                       @RequestParam(required = false, defaultValue = "1000") int max) {
+        return ResponseEntity.ok(studentService.getStudentsByAgeBetween(min, max));
+    }
+
+
     @PostMapping
+    @Operation(summary = "Create student")
     public ResponseEntity<Student> createStudent(@RequestBody Student student) {
         return ResponseEntity.ok(studentService.createStudent(student));
     }
 
     @PutMapping
+    @Operation(summary = "Edit student")
     public ResponseEntity<Student> editStudent(@RequestBody Student student) {
         return studentService.editStudent(student).isPresent() ?
                 ResponseEntity.ok(studentService.editStudent(student).get()) :
                 ResponseEntity.badRequest().build();
     }
 
     @DeleteMapping("{id}")
+    @Operation(summary = "Delete student by ID")
     public ResponseEntity<?> deleteStudent(@PathVariable("id") Long id) throws StudentNotFoundException {
         try {
             studentService.deleteStudent(id);
             return ResponseEntity.ok().build();
         } catch (StudentNotFoundException e) {
             return ResponseEntity.badRequest().build();
         }
+    }
 
+    @GetMapping("/faculty")
```

Comment:
здесь ошибка REST тоже - путь должен быть с id, потому что идёт обращение к ресурсу студента по id и у него уже берётся факультет
```suggestion
    @GetMapping("/{id}/faculty")
```